src/main/kotlin/no/fdk/dataset_catalog/controller/DatasetController.kt

```diff
@@ -20,12 +24,31 @@ class DatasetController(private val datasetService: DatasetService) {
             ?: ResponseEntity(HttpStatus.NOT_FOUND)
 
     @PostMapping(consumes = [MediaType.APPLICATION_JSON_VALUE], produces = [MediaType.APPLICATION_JSON_VALUE])
-    @Throws(FDKException::class)
     fun create(@PathVariable("catalogId") catalogId: String, @RequestBody dataset: Dataset): ResponseEntity<Unit> =
         try {
             datasetService.create(catalogId, dataset)
             ResponseEntity(HttpStatus.CREATED)
         } catch (e : Exception) {
             ResponseEntity(HttpStatus.BAD_REQUEST)
         }
-}
\ No newline at end of file
+
+    @PatchMapping(value = ["/{id}"], consumes = [MediaType.APPLICATION_JSON_VALUE], produces = [MediaType.APPLICATION_JSON_VALUE])
+    fun updateDataset(@PathVariable("catalogId") catalogId: String,
+                      @PathVariable id: String,
+                      @RequestBody patch: Dataset): ResponseEntity<Dataset> =
+        datasetService.updateDataset(catalogId, id, patch)
+            ?.let {
```

Comment:
Når det bare er en linje så kan du skrive `?.let { ResponseEntity(it, HttpStatus.OK) }`, ser litt bedre ut :P